### PR TITLE
feat(remoteconfig): add remote config endpoints

### PR DESCRIPTION
**Pull request by avara1986**
# Description
- Add Remote Config endpoint (`/v0.7/config`)
- Add endpoint to create/update Remote Config payloads

# Motivation
ASM needs to add e2e tests without a backend

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -326,7 +326,7 @@ docker build --tag testagent .
 Run the tagged image:
 
 ```bash
-docker run --rm -v ${PWD}/snaps:/snapshots --publish 8126:8126 agent
+docker run --rm -v ${PWD}/snaps:/snapshots --publish 8126:8126 testagent
```
Comment (Kyle-Verhoog) [inline README.md:329]: nit: can you introduce this in a separate PR, good catch though!

File: ddapm_test_agent/agent.py [hunk 2]
```diff
@@ -107,7 +108,7 @@ async def session_token_middleware(request: Request, handler: _Handler) -> web.R
     return await handler(request)
 
 
-class Agent:
+class Agent(ResponsesMixin):
```
[the remark this answers is not in the document]
Comment (avara1986) [reply ddapm_test_agent/agent.py:111]: I created this mixin to extend this feature to other parts (Telemetry, traces, debugger, etc). But I like your approach too. I can use this and refactor later if we need more responses ;) thank you @Kyle-Verhoog

File: ddapm_test_agent/trace_snapshot.py [hunk 3]
```diff
@@ -134,7 +134,7 @@ def _match_traces(t1s: List[Trace], t2s: List[Trace]) -> List[Tuple[Trace, Trace
     matched_t1s = set()
     matched_t2s = set()
     matches: List[Tuple[Trace, Trace]] = []
-    for (t1_trace_id, t2_trace_id, score) in similarities:
+    for t1_trace_id, t2_trace_id, score in similarities:
```
[the remark this answers is not in the document]
Comment (avara1986) [reply ddapm_test_agent/trace_snapshot.py:137]: the pre-check isn't working without this change and

File: README.md [hunk 4]
```diff
@@ -283,6 +283,22 @@ Return stats that have been received by the agent for the given session token.
...
 
+## /test/session/responses/config (POST)
+Create a Remote Config payload to retrieve in endpoint `/v0.7/config`
+
+```
+curl -X POST 'http://0.0.0.0:8126/test/session/responses/config/path' -d '{"roots": ["eyJ....fX0="], "targets": "ey...19", "target_files": [{"path": "datadog/2/ASM_DATA/blocked_users/config", "raw": "eyJydWxlc19kYXRhIjogW119"}], "client_configs": ["datadog/2/ASM_DATA/blocked_users/config"]}'
+```
+
+## /test/session/responses/config/path (POST)
+Due to Remote Config payload is quite complicated, this endpoint works like `/test/session/responses/config (POST)`
```
Comment (Kyle-Verhoog) [inline README.md:294]:
```suggestion
Due to Remote Config payload being quite complicated, this endpoint works like `/test/session/responses/config (POST)` 
```

**Review by Kyle-Verhoog — APPROVED** [review]
Great, thanks 🙌 ! A future improvement would be to have the endpoints be aware of the session token so that the responses are scoped to the current session and don't affect other sessions. This would enable concurrent usage.

**Pull request merged**